cmd/thanos/tools_stream_metric.go

```diff
@@ -0,0 +1,167 @@
+// Copyright (c) The Thanos Authors.
+// Licensed under the Apache License 2.0.
+
+package main
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"strings"
+	"time"
+
+	"github.com/go-kit/log"
+	"github.com/go-kit/log/level"
+	"github.com/oklog/run"
+	"github.com/opentracing/opentracing-go"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/prometheus/model/labels"
+	"github.com/thanos-io/thanos/pkg/extkingpin"
+	"github.com/thanos-io/thanos/pkg/store/storepb"
+
+	"github.com/prometheus/prometheus/tsdb/chunkenc"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials/insecure"
+)
+
+type rawMetricConfig struct {
+	storeAddr            string
+	metric               string
+	hoursAgo             int
+	hours                int
+	skipChunks           bool
+	ignoreWarnings       bool
+	equalLabelMatcher    string
+	notEqualLabelMatcher string
+}
+
+func registerFlags(cmd extkingpin.FlagClause) *rawMetricConfig {
+	conf := &rawMetricConfig{}
+	cmd.Flag("store", "Thanos Store API gRPC endpoint").Default("localhost:10901").StringVar(&conf.storeAddr)
+	cmd.Flag("metric", "The metric name to stream time series for this metric").Default("up").StringVar(&conf.metric)
```

Comment:
Hmm up can be a bit of a bad query. Let's not set defaults here, and just error out if no metric names are specified.

Reply:
updated